libs2ecore/include/s2e/S2EExecutor.h

```diff
@@ -205,6 +227,10 @@ class S2EExecutor : public klee::Executor {
     void computeNewStateGuids(std::unordered_map<klee::ExecutionState *, uint64_t> &newIds, klee::StateSet &parentSet,
                               klee::StateSet &childSet);
 
+    // If `condition` is a nullptr, then no path constraints will be added.
+    StatePair doFork(klee::ExecutionState &current, const klee::ref<klee::Expr> *condition,
```

Comment:
I would keep the old fork as is, don't modify it. It's specifically designed for the case where you have constraints. Adding one more special case will make it even more complex.